singlem/chainsaw.py

```diff
@@ -40,9 +43,57 @@ def chainsaw(**kwargs):
         # mkdir refpkg folder
         refpkg_path = os.path.join(graftm_path, os.path.basename(input_spkg.graftm_package().reference_package_path()))
         os.mkdir(refpkg_path)
-        shutil.copyfile(
-            os.path.join(input_spkg.graftm_package().reference_package_path(), 'CONTENTS.json'),
-            os.path.join(refpkg_path,'CONTENTS.json'))
+        # Change the seqinfo adding the prefix
+        seqinfo_basename = os.path.basename(input_spkg.graftm_package().taxtastic_seqinfo_path())
+        seqnames = []
+        with open(os.path.join(refpkg_path,seqinfo_basename),'w') as out:
+            first = True
+            for line in open(input_spkg.graftm_package().taxtastic_seqinfo_path()):
+                if first:
+                    out.write(line)
+                    first = False
+                else:
+                    # removing the "GCA_" check fixes a (hopefully) rare bug but probably makes an already horribly inefficient code section way worse 
+                    if keep_tree and "GCA_" not in line:
+                        seqnames.append(line.split(',')[0])
+                    out.write("{}{}".format(sequence_prefix,line))
+        if keep_tree:
+            for filename in os.listdir(os.path.join(input_spkg.graftm_package().reference_package_path())):
+                shutil.copyfile(
+                    os.path.join(input_spkg.graftm_package().reference_package_path(), filename),
+                    os.path.join(refpkg_path, filename))
+            # Add sequence prefix to tree and tree.log
+            logging.info("Adding sequence prefix to refpkg contents")
```

Comment:
When running without changing the sequence_prefix, this log statement isn't true. Can just be moved down to after the next `if`